### DISCUSSION TITLE
automatically adding Systems / Endpoints to findings

### DISCUSSION THREAD

**Discussion by hippyjm** [DefectDojo/django-DefectDojo#10315] [Q&A] [answered]
Good morning. this is potentially a very obvious one but cant see this in documentation (potentially looking in the wrong place)
Importing finding and endpoints separately works great. but it seems that for each finding i need to manually edit the finding to add the endpoint. or for each endpoint i can manually add the findings. is there a field im missing somewhere? im using the generic findings import at the moment. 
end goal is that during the import of findings is that i can automatically assign all the endpoints to each finding. 
![Screenshot 2024-06-03 102035](https://github.com/DefectDojo/django-DefectDojo/assets/137270373/884882de-5618-4fbc-9715-02d99f3a64af)

am i being daft here? or can someone point me to the right doc? 

in addition.. whats the difference between endpoint and host? when importing it populates both. had a look but no solid answer. 
still at the beginning of our journey with defect dojo and loving it so far

**Comment by mtesauro** [comment, marked as answer]
If you're using the generic importer, it depends if you're importing JSON or CSV.  Quickly skimming both parsers at https://github.com/DefectDojo/django-DefectDojo/tree/master/dojo/tools/generic it appears that the JSON supports endpoints and CSV does not.

So, if you're doing CSV imports, getting endpoints added during import won't happen but can for JSON - there might be examples of the JSON with endpoints here: https://github.com/DefectDojo/django-DefectDojo/tree/master/unittests/scans/generic

HTH

**Reply by hippyjm** [reply]
thanks. just found the format in one of the examples provided - generic_report3.json. working now. thank you
